docs/guide/ssr.md

```diff
@@ -215,6 +215,10 @@ If a dependency needs to be transformed by Vite's pipeline, for example, because
 
 For linked dependencies, they are not externalized by default to take advantage of Vite's HMR. If this isn't desired, for example, to test dependencies as if they aren't linked, you can add it to [`ssr.external`](../config/ssr-options.md#ssr-external).
 
+:::warning Running plugins in SSR
+Plugins hooks are only executed for local files (starting with `.` or `/`) during SSR, if you want to run plugins on `node_modules` files you should resolve them to a path starting with `/` in the `resolveId` hook.
+:::
```

Comment:
I think the phrasing could be interpreted differently than intended, e.g. if you want to run plugins on `node_modules` files, you should put that dependency in `ssr.noExternal`, which this is already explained above.

I'm not sure how to re-phrase this. You don't need to resolve to a path starting with `/`, virtual paths like `\0virtual:something` works too. The issue you hit specifically is with returning `https://...`, but that's very specific and not a common thing to do in plugins to document 🤔

Reply:
ok so the problems was that Vite does not add `/@id/` to imports starting with `https`, i misunderstood what the issue was